modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -325,6 +326,32 @@ public PartitionReplicaListener(
         cursors = new ConcurrentSkipListMap<>(IgniteUuid.globalOrderComparator());
 
         schemaCompatValidator = new SchemaCompatValidator(schemas, catalogTables);
+
+        placementDriver.listen(PrimaryReplicaEvent.PRIMARY_REPLICA_EXPIRED, (evt, e) -> {
+            if (!localNode.name().equals(evt.leaseholder())) {
+                return completedFuture(false);
+            }
+
+            LOG.info("Primary replica expired [grp={}]", replicationGroupId);
+
+            ArrayList<CompletableFuture<?>> futs = new ArrayList<>();
+
+            for (UUID txId : txCleanupReadyFutures.keySet()) {
```

Comment:
It could happens in the case when a expired, but the does not accepeted. I think again about it and desided to no trigger the expire event in the case.
Currently, we can change this warning message to just checkoing assertion.

Reply:
Rewrote the place to the `Map#compute` invocation. I hope it allw to no think about CME.

Reply:
Ok.